### PR TITLE
Usagov 257 permission for content editor role

### PR DESCRIPTION
**Pull request by oadekoya12**
## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-257

## Checklist for the Developer

- [x] A link to the JIRA ticket has been added above.
- [x] The JIRA ticket identifies the desired result of this change.
- [x] The JIRA ticket contains clear acceptance criteria.
- [x] The JIRA ticket contains clear testing steps.
- [x] The JIRA ticket contains a description of "Done".
- [x] Any preparation/installation/update steps required for this code to work properly (drush commands, scripts, configuration updates) are documented in the ticket.

## Checklist for the Peer Reviewers
- [x] The branch name of this PR matches the project standards.
- [x] QA steps are followed and any changes to process are updated in the ticket.
- [x] Code Standards are followed, there are no bad practices in use.
- [x] Config changes (if any) include only necessary changes.
- [x] No errors in Drupal or Client Browser.
- [x] Code has been tested locally (if possible).
- [ ] Following AC and Testing Steps verify changes work as expected.
- [ ] There are no known side-effects outside of expected behavior.

### CODE REVIEW

File: web/modules/custom/usa_workflow/usa_workflow.module [hunk 1]
```diff
@@ -1 +1,90 @@
...
+    'entity.node.edit_form',
+    'entity.node.latest_version'
+  ];
+
+  // Compare current logged in user and revised user are not the same
+  // also make sure the current logged in user as
+  // 'usa delete own content' permission.
+  if (in_array($route_name, $routes)) {
+    dpm($wfusers);
+    dpm($wfusers['currentUser']['usadeleteowncontent'] == FALSE);
```
Comment (ednark) [inline web/modules/custom/usa_workflow/usa_workflow.module]: Remove debug dpm() statements

File: web/modules/custom/usa_workflow/usa_workflow.module [hunk 2]
```diff
@@ -1 +1,90 @@
...
+  // also make sure the current logged in user as
+  // 'usa approve own content' permission.
+  if ((in_array($form_id, $newform)
+      || $wfusers['currentUser']['id'] === $wfusers['reviseduser']['id'])
+      && $wfusers['currentUser']['usaapproveowncontent'] === FALSE) {
+    unset($form['moderation_state']['widget'][0]['state']['#options']['published']);
+    if (isset($form['new_state'])) {
+      unset($form['new_state']['#options']['published']);
+    }
+    if (isset($form['actions'])) {
```
Comment (ednark) [inline web/modules/custom/usa_workflow/usa_workflow.module]: It looks like this should be wrapped in a 'usadeleteowncontent' check like the other function

**Review by ednark — CHANGES_REQUESTED** [review]

File: web/modules/custom/usa_workflow/usa_workflow.module (hunk 2 above)
Comment (oadekoya12) [reply web/modules/custom/usa_workflow/usa_workflow.module]: Do you want me to wrap it now?

Comment (ednark) [reply web/modules/custom/usa_workflow/usa_workflow.module]: This current logic is wrong. It is removing Deletion capability based off the Approval permission and not the Deletion permission. Please update the logic to something like this and remove it from the current if statment it is in.

`  
    if (isset($form['actions']) && $wfusers['currentUser']['usadeleteowncontent'] === FALSE) {
      unset($form['actions']['delete']);
    }
`

Comment (oadekoya12) [reply web/modules/custom/usa_workflow/usa_workflow.module]: Done

**Review by ednark — APPROVED** [review]

**Pull request merged**